opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
```

Comment:
this int, float be consolidated into Number here?

Reply:
If I use `Number` rather than `int, float` here, I will need to change it to:

`if not isinstance(value, (bool, str, list, tuple)) and not issubclass(type(value), Number):`

I think just using `int, float` here is more concise, but I can understand the appeal of being consistent with what is done in `_check_attribute_value_sequence`. What do you think?

Reply:
isinstance seems to work fine:

```
yusuket:/home/yusuket/zillow/datadog-usage-monitors (0) (master)
$ python -c "from numbers import Number; print(isinstance(1, (bool, Number)))"
True
```

Reply:
Similarly, please consider using `collections.abc.Sequence` instead of `list, tuple`.

Reply:
And if using `Number` here, also use it in the `AttributeValue` type for consistency (and note that this allows `Decimal` and `Fraction` numbers too).

Reply:
@toumorokoshi I'm not sure why my tests were failing after originally making this change. Just tried this again and it works.